src/differentiation/compute_hessian_ad.jl

```diff
@@ -96,3 +96,74 @@ function numauto_color_hessian(f,
     numauto_color_hessian!(H, f, x, hes_cache)
     return H
 end
+
+
+
+## autoauto_color_hessian
+
+mutable struct ForwardAutoColorHesCache{TJC,TG,TS,TC}
+    jac_cache::TJC
+    grad!::TG
+    sparsity::TS
+    colorvec::TC
+end
+
+function ForwardAutoColorHesCache(f,
+    x::AbstractVector{V},
+    colorvec::AbstractVector{<:Integer}=eachindex(x),
+    sparsity::Union{AbstractMatrix,Nothing}=nothing) where V
+
+    if sparsity === nothing
+        sparsity = sparse(ones(length(x), length(x)))
+    end
+
+    tag = ForwardDiff.Tag(f, V)
```

Comment:
Can we just directly use a package tag instead of a function tag? I don't think throwing the whole function is necessary, and because of the eltype of `V` it's still safe for nesting.